controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -259,6 +274,33 @@ func (r *BareMetalHostReconciler) Reconcile(request ctrl.Request) (result ctrl.R
 	return
 }
 
+// Consume inspect.metal3.io/hardwaredetails when either
+// inspect.metal3.io=disabled or there are no existing HardwareDetails
+func (r *BareMetalHostReconciler) updateHardwareDetails(request ctrl.Request, host *metal3v1alpha1.BareMetalHost) (bool, error) {
+	reqLogger := r.Log.WithValues("baremetalhost", request.NamespacedName)
```

Comment:
nit: a `baremetalhost` logger should be already available in the reconcile loop, could be passed as parameter (could save some cycles in all the cases where the logger is not needed)

Reply:
Yeah I did consider that, but wanted to avoid bloating the parameter list, and several other functions already do it this way.

How about I move this inside the conditional, so it's only done when the logger is needed?

Reply:
Yep, or in alternative it can be logged outside the function when `hwdUpdated` is set